systemFlake.nix

```diff
@@ -149,7 +150,7 @@ let
             ];
           })
         ] ++ host.modules;
-        specialArgs = host.specialArgs;
+        specialArgs = host.specialArgs // { channel = channelInfo; };
       } // (optionalAttrs (host.output == "nixosConfigurations") {
         inherit lib baseModules specialArgs;
```

Comment:
is this even working? Is it not getting overwritten by `host.output == "nixosConfigurations"` part? I might be misreading something.

Also, maybe it would make more sense to attach the name somewhere around here? https://github.com/gytis-ivaskevicius/flake-utils-plus/blob/6e6b93660f163e4fcb7f82f941c7733b3b200971/systemFlake.nix#L166-L173

Reply:
the specialArgs above has a `// host.specialArgs`. So theres no issue with specialArgs right now, but channel would get overwritten.

And yeah thats a good idea, I'll move `name` down.

Reply:
I found that nuance a bit implicit. Glad to know we bump on it.

Reply:
Ohh that doesn't actually work, since we can't pass a nixpkgs, I was just hoping to pass channel metadata, so its pulled directly from the `channels` argument.

Reply:
So I decided to go ahead and pass the instantiated nixpkgs and then added channel information - name and input - implicitly to all channels